components/core/src/util/mod.rs

```diff
@@ -70,3 +71,63 @@ where
 {
     s.serialize_str(&t.to_string())
 }
+
+/// Provide a way to convert numeric types safely to i64
+trait ToI64 {
+    fn to_i64(self) -> i64;
+}
+
+impl ToI64 for usize {
+    fn to_i64(self) -> i64 {
+        if mem::size_of::<usize>() >= mem::size_of::<i64>() && self > i64::max_value() as usize {
+            error!(
+                "Tried to convert an out-of-range usize ({}) to i64; using i64::max_value()",
+                self
+            );
+            i64::max_value()
+        } else {
+            self as i64
+        }
+    }
+}
+
+impl ToI64 for u64 {
+    fn to_i64(self) -> i64 {
+        if mem::size_of::<u64>() >= mem::size_of::<i64>() && self > i64::max_value() as u64 {
```

Comment:
There's no need to test the relative sizes of `u64` and `i64`; unlike `usize`, [they're intrinsic to the type](http://doc.rust-lang.org/1.31.0/std/mem/fn.size_of.html).
```suggestion
        if self > i64::max_value() as u64 {
```

Reply:
Ah, of course.  I just copy/pasted and spaced on that.